### DISCUSSION TITLE
migration from a normal postgresql installation

### DISCUSSION THREAD

**Discussion by onichandame** [tensorchord/pgvecto.rs#479] [Q&A] [answered]
Hi, I am going to install immich on my home server, on which a normal postgresql container is already running. I'd like to upgrade my installation of pg to the one with pgvecto.rs. How can I achieve this?

**Comment by gaocegege** [comment]
cc @VoVAllen

**Comment by VoVAllen** [comment]
Is your local postgres the standard one or customized?

**Reply by onichandame** [reply]
my pg is deployed using bitnami's helm chart, which uses the image https://hub.docker.com/r/bitnami/postgresql . the image itself is very close to the official one, so I'd consider it standard.

Just to clarify my question. There are 2 possible approaches that I can imagine:
1. replace the image with the one with pgvecto.rs while keeping the data directory
2. pg_dumpall from the standard installation and restore the data to the new one with pgvecto.rs

the first approach is less of hassle for the end users like me, but it may break the data if something goes wrong(version mismatch etc.). the second approach looks more promising to me, but I couldn't find any documentation to know whether it would work. An official recommendation is all I want.

**Reply by VoVAllen** [reply, marked as answer]
As long as the postgres major version is consistent, I think method 1 should work. That's how immich migrate from previous postgres image to pgvecto.rs. However, some of immich user met the problem due to locale setting so that the index need to be rebuilt. But no data is lost in the migration process
